packages/datadog-plugin-jest/src/index.js

```diff
@@ -218,6 +231,153 @@ function createHandleTestEvent (tracer, testEnvironmentMetadata, instrumenter) {
   }
 }
 
+function createWrapIt (tracer, globalConfig, globalInput, testEnvironmentMetadata) {
+  return function wrapIt (it) {
+    return function itWithTrace (description, specFunction, timeout) {
+      let oldSpecFunction = specFunction
+      if (specFunction.length) {
+        oldSpecFunction = promisify(oldSpecFunction)
+      }
+
+      const { childOf, commonSpanTags } = getTestSpanTags(tracer, testEnvironmentMetadata)
+
+      const testSuite = globalInput.jasmine.testPath.replace(`${globalConfig.rootDir}/`, '')
+
+      const newSpecFunction = tracer.wrap(
+        'jest.test',
+        {
+          type: 'test',
+          childOf,
+          tags: { ...commonSpanTags, [TEST_SUITE]: testSuite }
+        },
+        async (done) => {
+          const testSpan = tracer.scope().active()
+          const { currentTestName } = globalInput.expect.getState()
+          const resource = `${testSuite}.${currentTestName}`
+          testSpan.setTag(TEST_NAME, currentTestName)
+          testSpan.setTag(RESOURCE_NAME, resource)
+          testSpan.context()._trace.origin = CI_APP_ORIGIN
+          let result
+          globalInput.jasmine.testSpanByTestName[currentTestName] = testSpan
```

Comment:
Wouldn't that cause a memory leak?

Reply:
this global is scoped to the test file, and tore down afterwards